client/src/components/AppBar.js

```diff
@@ -0,0 +1,165 @@
+import React, { Component } from 'react';
+import MenuIcon from '@material-ui/icons/Menu';
+import Home from '@material-ui/icons/Home';
+import AssignmentTurnedIn from '@material-ui/icons/AssignmentTurnedIn'; 
+import DirectionsRun from '@material-ui/icons/DirectionsRun';
+import ArrowBack from '@material-ui/icons/ArrowBack';
+import Check from '@material-ui/icons/Check';
+import { withStyles } from '@material-ui/core/styles';
+import { withRouter } from 'react-router-dom';
+import PropTypes from 'prop-types';
+
+import {
+  AppBar,
+  Toolbar,
+  IconButton,
+  Drawer,
+  List,
+  Typography,
+  ListItemIcon,
+  ListItemText,
+  ListItem,
+  Divider,
+} from '@material-ui/core';
+
+const styles = {
+  container: {
+    position: 'fixed',
+    width: '100%',
+    left: '0',
+    top: '0',
+  },
+  space: {
+    height: '60px',
+  },
+  leftButton: {
+    marginLeft: -12,
+    marginRight: 20,
+  },
+  listItem: {
+    width: 250,
+  },
+  projectName: {
+    padding: 10,
+    marginBottom: 40,
+  },
+  title: {
+    position: 'absolute',
+    left: '50%',
+    transform: 'translate(-50%)',
+  },
+  rightButton: {
+    position: 'absolute',
+    right: 15,
+  },
+};
+
+class CustomAppBar extends Component {
+
+  constructor(props) {
+    super(props);
+
+    this.state = {
+      navPaneOpened: false,
+    }
+
+    this.handleCloseNavPane = this.handleCloseNavPane.bind(this);
+    this.handleOpenNavPane = this.handleOpenNavPane.bind(this);
+    this.handleBack = this.handleBack.bind(this);
+  }
+
+  handleCloseNavPane() {
+    this.setState({
+      navPaneOpened: false,
+    });
+  }
+
+  
+  handleOpenNavPane() {
+    this.setState({
+      navPaneOpened: true,
+    });
+  }
+  
+  handleBack() {
+    this.props.history.goBack();
```

Comment:
history nggak ada di propTypes, ini built in function ya? Lo nggak pass props dari Parent Componentnya kan